compiler/test/dotty/tools/repl/ReplCompilerTests.scala

```diff
@@ -347,6 +347,15 @@ class ReplCompilerTests extends ReplTest:
     assertEquals("java.lang.AssertionError: assertion failed", all.head)
   }
 
+  @Test def `i17333 print null result of toString`: Unit =
+    initially:
+      run("val tpolecat = new Object { override def toString(): String = null }")
+    .andThen:
+      assertEquals("val tpolecat: Object = null // non-null reference has null-valued toString", lines().head)
```

Comment:
I think this is easier to read, what do you think?
```suggestion
      assertEquals("val tpolecat: Object = null // .toString() was null", lines().head)
```

Reply:
I agree. I wondered even, `return value of .toString() was null` after the NPE message
```
scala> X.toString.toString
java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "X$.toString()" is null
```
and now looking at it, I wonder if it is more useful and less redundant to use the default toString
```
scala> X
val res5: X.type = X$@23ba2c1e // return value of "X$.toString()" is null
```
or similar.